File: app/src/main/java/com/example/review/New/ReviewList.java
package com.example.review.New;

import java.io.*;

public class ReviewList extends ArrayStoreList<ReviewStruct> {

    //关联指定的库
    public void connectOf(LibraryList libraries) {
        for (int i = 0; i < this.size(); i++) {
            ReviewStruct rs = get(i);

            int index = i * 2;
            rs.show  = libraries.get(index + 1);
            rs.match = libraries.get(index);
        }
    }

    //取字节流
    @Override
    public void toBytes(DataOutputStream dos) throws IOException {
        dos.writeInt(size());
        for (ReviewStruct libraryStruct : this) {
            byte[] bytes = libraryStruct.toBytes();
            dos.writeInt(bytes.length);
            dos.write(bytes);
        }
    }

    //写字节流
    @Override
    public void loadWith(DataInputStream dis) throws IOException {
        int size = dis.readInt();
        for (int i = 0; i < size; i++) {
            int    len   = dis.readInt();
            byte[] bytes = new byte[len];
            dis.read(bytes);
            ReviewStruct libraryStruct = new ReviewStruct(bytes);
            add(libraryStruct);
        }
    }
}

File: app/src/main/java/com/example/review/New/ReviewList.kt
package com.example.review.New

import java.io.DataInputStream
import java.io.DataOutputStream
import java.io.IOException

open class ReviewList : ArrayStoreList<ReviewStruct>() {
    //关联指定的库
    fun connectOf(libraries: LibraryList) {
        for (index in this.indices) {
            val rs = get(index)
            val index = index * 2
            rs.show = libraries[index + 1]
            rs.match = libraries[index]
        }
    }

    //取字节流
    @Throws(IOException::class)
    override fun toBytes(dos: DataOutputStream) {
        dos.writeInt(size)
        for (rl in this) {
            val bytes = rl.toBytes()
            dos.writeInt(bytes.size)
            dos.write(bytes)
        }
    }

    //写字节流
    @Throws(IOException::class)
    override fun loadWith(dis: DataInputStream) {
        val size = dis.readInt()
        for (i in 0 until size) {
            val len = dis.readInt()
            val bytes = ByteArray(len)
            dis.read(bytes)
            val rs = ReviewStruct(bytes)
            add(rs)
        }
    }
}